app/scripts/modules/core/src/pipeline/config/stages/evaluateVariables/EvaluateVariablesStageConfig.tsx

```diff
@@ -0,0 +1,75 @@
+import { IStageConfigProps } from 'core/pipeline/config/stages/core/IStageConfigProps';
+import * as React from 'react';
+import { set } from 'lodash';
+
+import { StageConfigField } from '../core/stageConfigField/StageConfigField';
+import { IStage } from 'core/domain';
+import { MapEditor } from '@spinnaker/core';
+
+export interface IEvaluateVariablesStageConfigState {
+  variables: any;
+}
+
+export class EvaluateVariablesStageConfig extends React.Component<
+  IStageConfigProps,
+  IEvaluateVariablesStageConfigState
+> {
+  public static getDerivedStateFromProps(props: IStageConfigProps): IEvaluateVariablesStageConfigState {
```

Comment:
I assume you have read https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html ?
I there a better approach?

Reply:
I guess we don't need state at all. It's just passing everything through to `MapEditor` anyways.